plugins/systems/behavior/Behavior.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+** EPITECH PROJECT, 2024
+** RType
+** File description:
+** Behavior.hpp file
+*/
+
+#ifndef BEHAVIOR_HPP
+    #define BEHAVIOR_HPP
+
+    #include <libconfig.h++>
+    #include "systems/ASystem.hpp"
+    #include "components/velocity/Velocity.hpp"
+    #include "components/position/Position.hpp"
+    #include "components/collider/Collider.hpp"
+    #include "components/damage/Damage.hpp"
+    #include "components/type/Type.hpp"
+    #include "components/spriteId/SpriteID.hpp"
+    #include "components/deathRange/DeathRange.hpp"
+    #include "components/gun/Gun.hpp"
+    #include "components/Ai/Ai.hpp"
+
+namespace Engine {
+    class GameEngine;
+}
+
+namespace Systems {
+    /**
+     * @class BehaviorSystem
+     * 
+     * @brief System class from Systems that handles the movement and attacks of non controllable entities.
+     * 
+     * This system is responsible for handling the movement and attacks of non playable entities.
+     */
+    class BehaviorSystem : public ASystem {
```

Comment:
Could you add doxygen comments to this class member functions ?

Reply:
Doxygen documentation has been added to the 4 member functions.

Reply:
Thanks